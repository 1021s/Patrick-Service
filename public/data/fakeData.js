module.exports = {
  interiorDetails: {
    bedroomsAndBathrooms: {
      bedrooms: 5, bathrooms: 3, fullBathrooms: 3, halfBathrooms: 0,
    },
    basement: { basement: 'error est nemo' },
    flooring: { flooring: 'qui facilis omnis' },
    heating: { heating: 'et dolores asperiores, aut quis porro' },
    appliances: { appliances: 'maxime, impedit, et, blanditiis, optio' },
    otherInteriorFeatures: { livableArea: '2930 sqft' },
  },
  propertyDetails: {
    parking: { parkingFeatures: 'sint qui debitis', garageSpaces: 'doloremque minima aut' }, property: { stories: '1', exteriorFeatures: 'praesentium', viewDescription: 'soluta, perferendis, quod' }, lot: { lotSize: '7331 acres' }, otherPropertyInformation: { parcelNumber: '70555' },
  },
  constructionDetails: { typeAndStyle: { homeType: 'deleniti laboriosam dolore' }, materialInformation: { roof: 'harum atque dolorum' }, condition: { newConstruction: 'false', yearBuilt: '1983' } },
  utilitiesGreenEnergyDetails: { utility: { sewerInformation: 'laudantium temporibus qui', internetAndTv: 'sint beatae odio' }, greenEnergy: { sunscore: 'dolores voluptatem excepturi: 1' } },
  communityAndNeighborhoodDetails: { location: { region: 'Buckinghamshire' } },
  hoaAndFinancialDetails: { otherFinancialInformation: { taxAssessedValue: '$2655956', annualTaxAmount: '$36073' } },
  other: {
    sourceDetails: { mlsId: '2078077' },
    otherFacts: {
      busLineNearby: 'false', propertyType: 'recusandae', buildingInfo: 'saepe id incidunt', energySource: 'ut vel repudiandae', masterBedroomLocation: 'voluptates', waterSource: 'temporibus', foundation: 'nobis consequatur reiciendis', possesion: 'porro', parkingType: 'ullam autem illum', form17: 'aspernatur', appliancesThatStay: 'aut officiis adipisci', heatingAndCooling: 'blanditiis molestiae aut', entranceLevel: 'iusto', roof: 'ea', utilityRoomLocation: 'aut', features: 'velit facilis harum', lotDetails: 'aut aliquid cumque', floorCovering: 'vel voluptas sed', style: 'illo soluta aliquid', potentialTerms: 'reprehenderit ex id', kitchenLocation: 'velit', architecture: 'voluptas', lotTopography: 'illo', exterior: 'voluptate', siteFeatures: 'et odit repellat', offers: 'minus', commission: '4',
    },
  },
  _id: '5deb33a4a3a4fb338d482387',
  listingId: '067',
  type: 'qui amet provident',
  yearBuilt: 2008,
  heating: 'aut provident architecto',
  cooling: 'reprehenderit dolorem voluptates',
  parking: 'neque autem enim',
  lot: '3653 sqft',
  priceSqft: '$780',
  __v: 0,
};
