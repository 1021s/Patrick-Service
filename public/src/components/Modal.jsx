/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable react/prop-types */
import React from 'react';

class Modal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      sourcesModal: false,
    };

    this.setWrapperRef = this.setWrapperRef.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.handleSecondModal = this.handleSecondModal.bind(this);
  }

  componentDidMount() {
    document.body.classList.add('body-modal');
    document.addEventListener('mousedown', this.handleClick, false);
  }

  componentWillUnmount() {
    document.body.classList.remove('body-modal');
    document.removeEventListener('mousedown', this.handleClick, false);
  }

  setWrapperRef(node) {
    this.wrapperRef = node;
  }

  handleClick(event) {
    const { toggleModal } = this.props;
    event.preventDefault();
    const { sourcesModal } = this.state;
    if ((!this.wrapperRef.contains(event.target)
      && this.wrapperRef.className === 'second-modal')
      || (sourcesModal && event.target.value === 'X')) {
      this.handleSecondModal(event);
      toggleModal(event);
    } else if (!this.wrapperRef.contains(event.target)) {
      toggleModal(event);
    } else if (sourcesModal && event.target.innerText !== 'X'
    && event.target.id !== 'second') {
      this.handleSecondModal(event);
    }
  }

  handleSecondModal(event) {
    event.preventDefault();
    const { sourcesModal } = this.state;
    this.setState({
      sourcesModal: !sourcesModal,
    });
  }

  render() {
    const { toggleModal, listing } = this.props;
    const { sourcesModal } = this.state;
    if (!sourcesModal) {
      return (
        <div className="modal" ref={this.setWrapperRef}>
          <div className="modal-content">
            <div className="modal-title">
              Home Facts by Data Source
            </div>
            <div className="grid-container-one">
              <span className="grid-item">
                &#8203;
              </span>
              <span className="grid-item">
                ALL SOURCES COMBINED &#8203;
                <img src="/images/question.png" alt="" id="question" height="12px" width="12px" onClick={this.handleSecondModal} onKeyPress={this.handleSecondModal} />
              </span>
              <span className="grid-item">
                COUNTY RECORDS
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Beds:
              </span>
              <span className="grid-item">
                {listing.interiorDetails.bedroomsAndBathrooms.bedrooms}
              </span>
              <span className="grid-item">
                {listing.interiorDetails.bedroomsAndBathrooms.bedrooms}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Baths:
              </span>
              <span className="grid-item">
                {listing.interiorDetails.bedroomsAndBathrooms.bathrooms}
              </span>
              <span className="grid-item">
                {listing.interiorDetails.bedroomsAndBathrooms.bathrooms}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Sqft:
              </span>
              <span className="grid-item">
                {`${listing.lot} / ${listing.propertyDetails.lot.lotSize}`}
              </span>
              <span className="grid-item">
                {`${listing.lot} / ${listing.propertyDetails.lot.lotSize}`}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Type:
              </span>
              <span className="grid-item">
                {listing.type}
              </span>
              <span className="grid-item">
                {listing.type}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Year Built:
              </span>
              <span className="grid-item">
                {listing.yearBuilt}
              </span>
              <span className="grid-item">
                {listing.yearBuilt}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Last Sold:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Parking:
              </span>
              <span className="grid-item">
                {listing.parking}
              </span>
              <span className="grid-item">
                {listing.parking}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Cooling:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Heating:
              </span>
              <span className="grid-item">
                {listing.heating}
              </span>
              <span className="grid-item">
                {listing.heating}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Fireplace:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Days on Zillow:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                MLS #:
              </span>
              <span className="grid-item">
                {listing.other.sourceDetails.mlsId}
              </span>
              <span className="grid-item">
                {listing.other.sourceDetails.mlsId}
              </span>
            </div>
            <div className="grid-container-two">
              <span className="grid-item" style={{ fontWeight: 500, fontSize: 23 }}>
                Other facts
              </span>
              <span className="grid-item" style={{ fontWeight: 500, fontSize: 23 }}>
                &#8203;
              </span>
              <span className="grid-item" style={{ fontWeight: 500, fontSize: 23 }}>
                &#8203;
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                More information:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
            </div>
          </div>
          <button className="close-modal" type="button" onClick={toggleModal}>X</button>
        </div>
      );
    }
    return (
      <div ref={this.setWrapperRef}>
        <div className="second-modal" id="second">
          <div className="second-modal-content" id="second">
            <div id="second">DISCLAIMER</div>
            <div id="second">
              The information in this column combines data from broker listing feeds,
              user-submitted information, and county records.
              The information in the &apos;county records&apos;
              column contains only what is recorded in county records.
            </div>
            <button className="close-second-modal" type="button" id="second" onClick={this.handleSecondModal}>X</button>
          </div>
        </div>
        <div className="modal">
          <div className="modal-content">
            <div className="modal-title">
              Home Facts by Data Source
            </div>
            <div className="grid-container-one">
              <span className="grid-item">
                &#8203;
              </span>
              <span className="grid-item">
                ALL SOURCES COMBINED &#8203;
                <img src="/images/question.png" alt="" id="question" height="12px" width="12px" onClick={this.handleSecondModal} onKeyPress={this.handleSecondModal} />
              </span>
              <span className="grid-item">
                COUNTY RECORDS
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Beds:
              </span>
              <span className="grid-item">
                {listing.interiorDetails.bedroomsAndBathrooms.bedrooms}
              </span>
              <span className="grid-item">
                {listing.interiorDetails.bedroomsAndBathrooms.bedrooms}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Baths:
              </span>
              <span className="grid-item">
                {listing.interiorDetails.bedroomsAndBathrooms.bathrooms}
              </span>
              <span className="grid-item">
                {listing.interiorDetails.bedroomsAndBathrooms.bathrooms}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Sqft:
              </span>
              <span className="grid-item">
                {`${listing.lot} / ${listing.propertyDetails.lot.lotSize}`}
              </span>
              <span className="grid-item">
                {`${listing.lot} / ${listing.propertyDetails.lot.lotSize}`}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Type:
              </span>
              <span className="grid-item">
                {listing.type}
              </span>
              <span className="grid-item">
                {listing.type}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Year Built:
              </span>
              <span className="grid-item">
                {listing.yearBuilt}
              </span>
              <span className="grid-item">
                {listing.yearBuilt}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Last Sold:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Parking:
              </span>
              <span className="grid-item">
                {listing.parking}
              </span>
              <span className="grid-item">
                {listing.parking}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Cooling:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Heating:
              </span>
              <span className="grid-item">
                {listing.heating}
              </span>
              <span className="grid-item">
                {listing.heating}
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Fireplace:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                Days on Zillow:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                MLS #:
              </span>
              <span className="grid-item">
                {listing.other.sourceDetails.mlsId}
              </span>
              <span className="grid-item">
                {listing.other.sourceDetails.mlsId}
              </span>
            </div>
            <div className="grid-container-two">
              <span className="grid-item" style={{ fontWeight: 500, fontSize: 23 }}>
                Other facts
              </span>
              <span className="grid-item" style={{ fontWeight: 500, fontSize: 23 }}>
                &#8203;
              </span>
              <span className="grid-item" style={{ fontWeight: 500, fontSize: 23 }}>
                &#8203;
              </span>
              <span className="grid-item" style={{ fontWeight: 'bold' }}>
                More information:
              </span>
              <span className="grid-item">
                --
              </span>
              <span className="grid-item">
                --
              </span>
            </div>
          </div>
          <button className="close-modal" type="button" value="X" onClick={toggleModal}>X</button>
        </div>
      </div>
    );
  }
}

export default Modal;
