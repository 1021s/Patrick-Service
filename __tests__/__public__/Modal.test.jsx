/* eslint-env jest */
/* eslint-env node */
import 'jsdom-global/register';
import React from 'react';
import '../../setupTests';
import { mount, shallow } from 'enzyme';
import fakeData from '../../public/data/fakeData';
import Modal from '../../public/src/components/Modal';

describe('Modal render suite', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Modal listing={fakeData} debug />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const component = mount(<Modal listing={fakeData} />);
    expect(component).toMatchSnapshot();
  });
});

describe('Modal click suite', () => {
  it('button click should show second modal', () => {
    const wrapper = mount(<Modal listing={fakeData} />);
    wrapper
      .find('#question')
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('button click should hide second modal', () => {
    const wrapper = mount(<Modal listing={fakeData} />);
    wrapper
      .find('#question')
      .simulate('click')
      .find('#question')
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('button click should show second modal', () => {
    const wrapper = mount(<Modal listing={fakeData} />);
    wrapper
      .find('.close-modal')
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
