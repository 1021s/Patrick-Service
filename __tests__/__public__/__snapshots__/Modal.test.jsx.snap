// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal click suite button click should hide second modal 1`] = `ReactWrapper {}`;

exports[`Modal click suite button click should show second modal 1`] = `ReactWrapper {}`;

exports[`Modal click suite button click should show second modal 2`] = `ReactWrapper {}`;

exports[`Modal render suite should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Modal render suite should render correctly with props 1`] = `ReactWrapper {}`;
